src/test/java/GmailServletTest.java

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import com.google.api.services.gmail.model.Message;
+import com.google.sps.model.AuthenticationVerifier;
+import com.google.sps.model.GmailClient;
+import com.google.sps.model.GmailClientFactory;
+import com.google.sps.servlets.GmailServlet;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+/**
+ * Test Gmail Servlet to ensure response contains correctly parsed messageIds. Assumes
+ * AuthenticatedHttpServlet is functioning properly (those tests will fail otherwise).
+ */
+@RunWith(JUnit4.class)
+public final class GmailServletTest {
+  private static final AuthenticationVerifier authenticationVerifier =
+      Mockito.mock(AuthenticationVerifier.class);
+  private static final GmailClientFactory gmailClientFactory =
+      Mockito.mock(GmailClientFactory.class);
+  private static final GmailClient gmailClient = Mockito.mock(GmailClient.class);
+  private static final GmailServlet servlet =
+      new GmailServlet(authenticationVerifier, gmailClientFactory);
```

Comment:
I think it's safer to initialize all these fields inside `init()` and make them non-static for the reasons I mentioned in the previous PR (or inside `classInit` when appropriate to make it clear that they will not be changing throughout tests)

Reply:
Fixed!